### ISSUE TITLE
Running in loop takes up RAM spaces

### ISSUE THREAD

**Issue by zaowad** [LCAV/pyroomacoustics#116]
```
import torchaudio
import pyroomacoustics as pra
import numpy as np
from IPython.display import Audio
import rir_online_script as ros
import glob
from tqdm import tqdm

for i in tqdm(files):
    audio,sr=torchaudio.load(i)
    rir_audio=ros.generate_rir(audio[0].numpy(),sr,pra,0.7)
```
this is my code snippet to generate rir for multiple audios in a folder. I am not appending generated arrays into any list. But somehow it ends up taking ram space. The function generate_rir looks like this:
```
def generate_rir(signal,fs,pra,factor):
    x2=np.random.uniform(4,7)
    y2=np.random.uniform(3,5)

    #height=np.random.uniform(2,5)
    height=np.random.uniform(2,4)
    corners = np.array([[0,0], [x2,0], [x2,y2],[0,y2]]).T
    room = pra.Room.from_corners(corners, fs=fs, max_order=8, absorption=0.2)
    room.extrude(height)

    # add source and set the signal to WAV file content
    source_height=np.random.uniform(0.5,0.9)
    room.add_source([1., 1.,source_height], signal=signal)

    # add single-microphone array
    x,y,z=np.random.uniform(3.5,x2-1.5),np.random.uniform(2,y2-1),np.random.uniform(2,height-1.5)
    R = np.array([[x], [y], [z]])  # [[x], [y], [z]]
    room.add_microphone_array(pra.MicrophoneArray(R, room.fs))

    # compute image sources
    room.image_source_model(use_libroom=True)
    room.simulate()
    g=np.max(signal)/np.max(room.mic_array.signals[0,:])

    #k=np.random.uniform(0.75,1)
    return room.mic_array.signals[0,:]*g*factor 
```

It would be much help if you can get me around this problem.Testing the impact of rir in an ASR setup.

**Comment by fakufaku** [comment]
Thank you very much for reporting this. Approximately how many files, and of what length are you processing ? At what point does memory become full ?

**Comment by zaowad** [comment]
Thank you for your response mr @fakufaku . I am processing 75205 wav files.Longest audio file is 10  second and most of them ranges from 6 to 10 second.It starts taking up memory as soon as the loop starts(i am checking memory status with htop command) and takes almost 42 gb of memories after finishing just 30000 audio files.At first i was storing them up by running the code thrice.But to reduce  WER on validation set i need to perform random online rir augmentation.So it would be much help if u can find me a solution.

**Comment by fakufaku** [comment]
@zaowad Sorry it took so long. I could reproduce the issue locally with the following code.
```
import glob

import numpy as np

import pyroomacoustics as pra
# import torchaudio
# from tqdm import tqdm


def generate_rir(signal, fs, pra, factor):
    x2 = np.random.uniform(4, 7)
    y2 = np.random.uniform(3, 5)

    # height=np.random.uniform(2,5)
    height = np.random.uniform(2, 4)
    corners = np.array([[0, 0], [x2, 0], [x2, y2], [0, y2]]).T
    room = pra.Room.from_corners(corners, fs=fs, max_order=8, absorption=0.2)
    room.extrude(height)

    # add source and set the signal to WAV file content
    source_height = np.random.uniform(0.5, 0.9)
    room.add_source([1.0, 1.0, source_height], signal=signal)

    # add single-microphone array
    x, y, z = (
        np.random.uniform(3.5, x2 - 1.5),
        np.random.uniform(2, y2 - 1),
        np.random.uniform(2, height - 1.5),
    )
    R = np.array([[x], [y], [z]])  # [[x], [y], [z]]
    room.add_microphone_array(pra.MicrophoneArray(R, room.fs))

    # compute image sources
    room.image_source_model(use_libroom=True)
    room.simulate()
    g = np.max(signal) / np.max(room.mic_array.signals[0, :])

    # k=np.random.uniform(0.75,1)
    return room.mic_array.signals[0, :] * g * factor


# Try with random sequences first
for i in range(72000):
    if i % 500 == 0:
        print("Epoch", i)
    sr = 16000
    audio = np.random.randn(10 * sr)
    rir_audio = generate_rir(audio, sr, pra, 0.7)
```
This is a serious issue and I will look into it. It seems there is a memory leak somewhere!

**Comment by fakufaku** [comment]
I could find and fix the memory leak. Thanks for your help in finding it! The corrected version now lives in the branch [bugfix_issue117_cython](https://github.com/LCAV/pyroomacoustics/tree/bugfix_issue117_cython).

To fix your problem there is however an easy solution that doesn't require updating the package (the fixed will be available via pip in only a few days). You are simulating a rectangular room, but you are using the general simulator for polyhedral room. Rectangular (aka shoebox) rooms can be simulated much more efficiently and there is a dedicated class.
Just replace
```
height = np.random.uniform(2, 4)
corners = np.array([[0, 0], [x2, 0], [x2, y2], [0, y2]]).T
room = pra.Room.from_corners(corners, fs=fs, max_order=8, absorption=0.2)
room.extrude(height)
```
with
```
room = pra.ShoeBox([x2, y2, height], fs=fs, max_order=8, absorption=0.2)
```
and it should be both much faster, and not leak memroy.

**Comment by fakufaku** [comment]
Fixed in #122
